halo2_proofs/src/dev.rs

```diff
@@ -476,7 +491,7 @@ impl<F: Field> Assignment<F> for MockProver<F> {
     }
 }
 
-impl<F: Field + Ord> MockProver<F> {
+impl<F: PrimeField + Ord> MockProver<F> {
```

Comment:
This is a change to the public API. What in Dynamic Lookups requires it?

Reply:
The method `ConstraintSystem::lookup_dynamic` requires `F: From<u64>`, which is not available on the `ff::Field` trait.
```rust
    #[cfg(feature = "unstable-dynamic-lookups")]
    pub fn lookup_dynamic(
        &mut self,
        table: &DynamicTable,
        table_map: impl FnOnce(&mut VirtualCells<'_, F>) -> DynamicTableMap<F>,
    ) -> usize
    where
        F: PrimeField,
    {
        [...]
        table_map.push((
            selector.clone() * Expression::Constant(F::from(table.tag())),
            selector * Expression::VirtualColumn(VirtualColumn(*table)),
        ));
        [...]
    }
```

Reply:
I have a (slightly awkward) workaround that avoids changing this bound: 75dedfbef439397bcd320c5e6da1198b4321101d